questionnaire/submissions-service.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+const createMessageBusCaller = require('../services/message-bus');
+const createQuestionnaireDAL = require('./questionnaire-dal');
+
+function createSubmissionService({logger}) {
+    const db = createQuestionnaireDAL({logger});
+
+    async function getQuestionnaireIdsBySubmissionStatus(status) {
+        return db.getQuestionnaireIdsBySubmissionStatus(status);
+    }
+
+    async function updateQuestionnaireSubmissionStatus(questionnaireId, submissionStatus) {
+        return db.updateQuestionnaireSubmissionStatus(questionnaireId, submissionStatus);
+    }
+
+    async function postFailedSubmissions() {
+        const questionnaireIds = await getQuestionnaireIdsBySubmissionStatus('FAILED');
+        let response = [];
+        const messageBusCaller = createMessageBusCaller({logger});
+        const promises = questionnaireIds.map(async id => {
```

Comment:
What's the intent for these promises vs fire and forget?

Reply:
so that it is somewhat testable and so the coverage is not 0%